private/buf/cmd/buf/buf_test.go

```diff
@@ -1987,6 +1987,95 @@ func TestFormatExitCode(t *testing.T) {
 	assert.NotEmpty(t, stdout.String())
 }
 
+func TestFormatStdin(t *testing.T) {
+	stdin := bytes.NewBuffer(
+		[]byte(`
+syntax = "proto3";
+
+
+package foo;
+`),
+	)
+	stdout := bytes.NewBuffer(nil)
+	testRun(
+		t,
+		0,
+		stdin,
+		stdout,
+		"format",
+		"--stdin",
+	)
+	assert.Equal(
+		t,
+		`syntax = "proto3";
+
+package foo;
+`,
+		stdout.String(),
+	)
+
+	stdin = bytes.NewBuffer(
+		[]byte(`
+syntax = "proto3";
```

Comment:
Can we add a test for multiple files concat'ed together, and either expect success or expect error? If error, document that `--stdin` can only take one file